src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -229,8 +234,8 @@ private UserDetails retrieveUser(String username, UsernamePasswordAuthentication
     public UserDetails retrieveUser(final String username, final String password, final String domainName, final List<SocketInfo> ldapServers) {
         UserDetails userDetails;
         try {
-            userDetails = userCache.get(username, new Callable<UserDetails>() {
-                public UserDetails call() throws AuthenticationException {
+            userDetails = userCache.get(username, new Callable<ActiveDirectoryUserDetail>() {
```

Comment:
Can be a userDetail as you will have cacheMiss (which should be defined immediately above this line) with the type info